### PR TITLE
Feature: Support mattermost notifier

### PR DESCRIPTION
**Pull request by frezbo**
Signed-off-by: Noel Georgi <git@frezbo.dev>

### CODE REVIEW

File: search.go [hunk 1]
```diff
@@ -239,6 +242,39 @@ func getAvailableSessions(response []byte, age int) error {
...
+	}
+	if notifier == "mattermost" {
+		sendMatterMostMsg(buf.String())
+	}
+	return nil
+}
+
+func sendMatterMostMsg(message string) error {
+	apiToken := os.Getenv("MATTERMOST_API_TOKEN")
+	client := mattermost.NewAPIv4Client(os.Getenv("MATTERMOST_URL"))
```
Comment (PrasadG193) [inline search.go]: @frezbo it should also work if someone passes the creds with CLI flags

File: search.go [hunk 2]
```diff
@@ -239,6 +242,39 @@ func getAvailableSessions(response []byte, age int) error {
...
+	client.AuthToken = apiToken
+	client.AuthType = mattermost.HEADER_AUTH
+	client.HttpHeader = map[string]string{
+		"Authorization": fmt.Sprintf("Bearer %s", apiToken),
+	}
+	user, res := client.GetMe("")
+	if res.StatusCode != 200 {
+		return errors.New("Auth failure")
+	}
+	direct, res := client.CreateDirectChannel(user.Id, os.Getenv("MATTERMOST_USER_ID"))
```
Comment (PrasadG193) [inline search.go]: same here, it should be possible to provide userID as a CLI flag

File: README.md [hunk 3]
```diff
@@ -64,6 +65,12 @@ covaccine-notifier --state Maharashtra --district Akola --age 27  --email <email
 covaccine-notifier --pincode 444002 --age 27  --email <email-id> --password <email-password>
 ```
 
+#### Notifications to mattermost
+
+```
+ATTERMOST_URL="<mattermost url>" MATTERMOST_API_TOKEN="<mattermost bot api token>" MATTERMOST_USER_ID="<mattermost user id>" ./covaccine-notifier --state Maharashtra --district Akola --age 27 -n mattermost
```
Comment (PrasadG193) [inline README.md]: Please add an example with CLI flags. It should be possible to provide Mattermost creds with CLI args

**Comment by PrasadG193** [comment]
@frezbo Please run `go mod tidy`. Looks like go.sum is not tidyed

**Comment by PrasadG193** [comment]
FYI, I am working on refactoring notifier interface. We might need to do some changes accordingly 
https://github.com/PrasadG193/covaccine-notifier/pull/33#discussion_r636886459

**Comment by frezbo** [comment]
I'll create a new PR once #33 is merged. So that I can rebase based on the new notifier interface. Thanks :+1:

**Pull request closed without merging**